modules/orchestrator/scheduler/executor/plugins/k8s/deployment.go

```diff
@@ -737,7 +737,7 @@ func (k *Kubernetes) newDeployment(service *apistructs.Service, serviceGroup *ap
 	}
 	k.AddSpotEmptyDir(&deployment.Spec.Template.Spec)
 
-	if err = DereferenceEnvs(deployment); err != nil {
+	if err = DereferenceEnvs(&deployment.Spec.Template); err != nil {
```

Comment:
what different between `deployment.Spec.Template.Spec` and `deployment.Spec.Template`?

Reply:
deployment.Spec.Template.Spec is same spec with job， this DereferenceEnvs is used for deployment and job。